http-client-spi/src/main/java/software/amazon/awssdk/http/ContentStreamProvider.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.http;
+
+import java.io.InputStream;
+import software.amazon.awssdk.annotations.SdkPublicApi;
+
+/**
+ * Provides the content stream of a request.
+ * <p>
+ * Each call to to the {@link #reset()} method must result in a stream whose position is at the beginning of the content.
+ * Implementations are not required to return the same instance for each call, and may close or otherwise discard the stream
+ * returned from the previous invocation.
+ *
+ */
+@SdkPublicApi
+@FunctionalInterface
+public interface ContentStreamProvider {
+    /**
+     * @return The content stream.
+     */
+    InputStream reset();
```

Comment:
Can we name this something other than reset? reset makes it sound side-affecty rather than returning a new stream. getNewStream()? getStream()? something like that.

Reply:
done